server/src/main/java/org/jetbrains/bsp/bazel/server/sync/TargetKindResolver.java

```diff
@@ -7,22 +7,24 @@
 import org.jetbrains.bsp.bazel.server.sync.model.Tag;
 
 public class TargetKindResolver {
-  private final Map<String, Tag> ruleSuffixToTargetType =
-      Map.of(
-          "library", Tag.LIBRARY,
-          "binary", Tag.APPLICATION,
-          "test", Tag.TEST);
+    private final Map<String, Tag> ruleSuffixToTargetType =
+            Map.of(
+                    "library", Tag.LIBRARY,
+                    "binary", Tag.APPLICATION,
+                    "test", Tag.TEST);
 
-  public Set<Tag> resolveTags(TargetInfo targetInfo) {
-    var tag =
-        ruleSuffixToTargetType.entrySet().stream()
-            .filter(entry -> targetInfo.getKind().endsWith("_" + entry.getKey()))
-            .map(Map.Entry::getValue)
-            .findFirst()
-            .orElse(Tag.NO_IDE);
-    if (targetInfo.getTagsList().contains("no-ide")) {
-      return HashSet.of(tag, Tag.NO_IDE);
+    public Set<Tag> resolveTags(TargetInfo targetInfo) {
+        var tag =
+                ruleSuffixToTargetType.entrySet().stream()
+                        .filter(entry -> targetInfo.getKind().endsWith("_" + entry.getKey()))
+                        .map(Map.Entry::getValue)
+                        .findFirst()
+                        .orElse(Tag.NO_IDE);
+        if (targetInfo.getTagsList().contains("no-ide")) {
+            return HashSet.of(tag, Tag.NO_IDE);
+        } else if (targetInfo.getTagsList().contains("manual")) {
+            return HashSet.of(Tag.MANUAL);
```

Comment:
we would return `HashSet.of(tag, Tag.MANUAL)` since we dont want to lose information about type of the target - for example it's `_test` with `manual` flag - we want to keep all the info

Reply:
oh right, I changed it